rust/src/measurements/make_private_expr/expr_laplace/mod.rs

```diff
@@ -0,0 +1,333 @@
+    ) -> PolarsResult<Series>
+    where
+        // the physical (rust) dtype must be able to be converted into a big integer (on the input side)
+        for<'a> IBig: From<PT::Physical<'a>>,
+        // a big integer must be able to be converted into the physical (rust) dtype (on the output side)
+        for<'a> PT::Physical<'a>: SaturatingCast<IBig>,
+        // must be able to construct the chunked array corresponding to the physical dtype from an iterator
+        for<'a> PT::Array:
+            ArrayFromIter<PT::Physical<'a>> + ArrayFromIter<Option<PT::Physical<'a>>>,
+        // must be able to convert the chunked array to a series
+        ChunkedArray<PT>: IntoSeries,
+    {
+        let scale = RBig::try_from(scale)
+            .map_err(|_| PolarsError::ComputeError("scale must be finite".into()))?;
+        Ok(series
+            // unpack the series into a chunked array
+            .unpack::<PT>()?
+            // apply Laplace noise to the non-null values
+            .try_apply_nonnull_values_generic(|v| {
+                let v = IBig::from(v);
+                sample_discrete_laplace(scale.clone())
+                    .map(|noise| PT::Physical::saturating_cast(v + noise))
+            })?
+            // convert the resulting chunked array back to a series
+            .into_series())
+    }
+
+    // PT stands for Polars Type
+    fn laplace_impl_float<PT: 'static + PolarsDataType>(
+        series: &Series,
+        scale: f64,
+    ) -> PolarsResult<Series>
+    where
+        // the physical (rust) dtype must be a float
+        for<'a> PT::Physical<'a>: Float + InfCast<f64>,
+        // for calibration of the discretization constant k
+        for<'a> i32: ExactIntCast<<PT::Physical<'a> as FloatBits>::Bits>,
+        // must be able to construct the chunked array corresponding to the physical dtype from an iterator
+        for<'a> PT::Array:
+            ArrayFromIter<PT::Physical<'a>> + ArrayFromIter<Option<PT::Physical<'a>>>,
+        // must be able to convert the chunked array to a series
+        ChunkedArray<PT>: IntoSeries,
+    {
+        // cast to the physical (rust) dtype (either f32 or f64)
+        let scale = PT::Physical::inf_cast(scale)?;
+        let k = get_discretization_consts::<PT::Physical<'static>>(None)?.0;
+        Ok(series
+            // unpack the series into a chunked array
+            .unpack::<PT>()?
+            // apply Laplace noise to the non-null values
+            .try_apply_nonnull_values_generic(|v| sample_discrete_laplace_Z2k(v, scale, k))?
+            // convert the resulting chunked array back to a series
+            .into_series())
+    }
+
+    use DataType::*;
+    match series.dtype() {
+        UInt32 => laplace_impl_integer::<UInt32Type>(series, scale),
+        UInt64 => laplace_impl_integer::<UInt64Type>(series, scale),
+        Int8 => laplace_impl_integer::<Int8Type>(series, scale),
+        Int16 => laplace_impl_integer::<Int16Type>(series, scale),
```

Comment:
but signed 8 and 16bit are supported? might be the case, but seems odd.

Reply:
Yep, just following along.